### PR TITLE
Fix detailed push documentation

### PR DESCRIPTION
**Pull request by jimmykarily**

### CODE REVIEW

File: docs/detailed-push-process.md [hunk 1]
```diff
@@ -5,53 +5,49 @@ But while doing so, it makes sure those components are deployed correctly and wo
...
-[a set of Tekton pipeline definitions](https://github.com/tektoncd/catalog/tree/main/task/buildpacks) that can be used to do exactly that.
-All we need to do is to tell the pipeline [where to find the application source](https://github.com/epinio/epinio/blob/bdcb1833829d58c5bdcf0e8ecb1998d766e72a13/embedded-files/tekton/triggers.yaml#L36-L44)
-and then trigger the pipeline to start.
+## 4. Trigger the pipeline
 
-Triggering happens using a [TriggerBinding](https://github.com/tektoncd/triggers/blob/main/docs/triggerbindings.md) to which Gitea will make a request.
-When you pushed your application in the previous step, Epinio set up a webhook on the Gitea project that makes a call to the trigger binding endpoint whenever
-the code is pushed to Gitea.
+When the Epinio API server receives the stage request, it will create a [`PipelineRun`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md) that will run the staging Tekton pipeline using the version of the code referenced in the request. This pipeline has 3 steps. Their role is described in the following 3 steps.
```
Comment (andreas-kupries) [inline docs/detailed-push-process.md:27]: `has 3 steps ... following 3 steps.` might be better phrased as
`has 3 steps ... following 3 sections.`

File: docs/detailed-push-process.md [hunk 2]
```diff
@@ -5,53 +5,49 @@ But while doing so, it makes sure those components are deployed correctly and wo
...
 
-So after you push your code, the above will ensure buildpacks will create an image for your application. You can see the logs of the staging process
-either by pushing the app with the `--verbosity 1` flag set, or by looking at the logs of the staging pods that appear in your cluster.
+## 5. Clone
 
-So an image is generated, but where is it stored? Read on.
+The first step of the staging Tekton pipeline is cloning the code from Gitea to a [workspace](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md). This makes the code available to the following steps.
```
Comment (andreas-kupries) [inline docs/detailed-push-process.md:31]: `is cloning` :arrow_right: `clones`

File: docs/detailed-push-process.md [hunk 3]
```diff
@@ -5,53 +5,49 @@ But while doing so, it makes sure those components are deployed correctly and wo
...
 
-## 3. The registry
+## 6. Stage
 
-Epinio installs a container registry inside your cluster and that is where the application images are stored. This was installed when you first did `epinio install` and is
-used to make the setup easier (by not having to configure an external registry) and staging faster (by keeping all image transferring local to the cluster).
+The second step of the staging Tekton pipeline is using the [paketo](buildpacks) to create an container image for your application. The definition of this Tekton task can be found [in the relevant upstream repo](https://github.com/tektoncd/catalog/tree/main/task/buildpacks/0.2) (though a copy of that is embedded in the Epinio binary).
```
Comment (andreas-kupries) [inline docs/detailed-push-process.md:35]: `is using` :arrow_right: `uses`
`an container` :arrow_right: `a container`

File: docs/detailed-push-process.md [hunk 4]
```diff
@@ -5,53 +5,49 @@ But while doing so, it makes sure those components are deployed correctly and wo
...
+The result of a successful staging process is a new image pushed to the Registry component of Epinio.
+
+This component is installed as part of the `epinio install` command and it is where the application images are stored. This is to make the setup easier (by not having to configure an external registry) and staging faster (by keeping all image transferring local to the cluster).
```
Comment (andreas-kupries) [inline docs/detailed-push-process.md:38]: `is to make` :arrow_right: `makes`

File: docs/detailed-push-process.md [hunk 5]
```diff
@@ -5,53 +5,49 @@ But while doing so, it makes sure those components are deployed correctly and wo
...
-You can read how these work in Kubernetes following the provided links but if you have to know one thing is that Ingress is the thing that describes how a request that uses the Application's url is routed to the application
-container. In Kubernetes, the thing that reads that description and implements the routing is called an Ingress Controller. Such an Ingress Controller is provided by [Traefik](https://doc.traefik.io/traefik/providers/kubernetes-ingress/).
+The last step of the staging Tekton pipeline is creating the runtime Kubernetes resources that are needed to make your application available to the users outside the Kubernetes cluster. The most important resources that are created are a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/), a [Service](https://kubernetes.io/docs/concepts/services-networking/service/) and an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resource. The Deployment is using the image that was pushed as part of the staging step (see arrows 6 and 8 on the image).
```
Comment (andreas-kupries) [inline docs/detailed-push-process.md:46]: `is creating` :arrow_right: `creates`

Question: `The most important resources are ...` implies that there are resources beyond Deployment, Service, and Ingress. What are these resources ?

File: docs/detailed-push-process.md [hunk 6]
```diff
@@ -63,16 +59,13 @@ myapplication.my_epinio_system_domain.com
...
 You can get the route of your application with `epinio apps list` or `epinio apps show myapplication`
 
-## 6. Additional things
+## Additional things
 
-During installation, if you specified a system domain using the `--system-domain` parameter, then your application routes will be sudomains of that domain.
-Epinio considers this domain to be a production server and thus creates a production level TLS certificate for your application using [Let's Encrypt](https://letsencrypt.org/).
-This happens by using the [cert-manager](https://cert-manager.io/docs/installation/kubernetes/) which is one more of the components Epinio installs with `epinio install`.
+During installation, if you specified a system domain using the `--system-domain` parameter, then your application routes will be subdomains of that domain.
+Epinio considers this domain to be a production server and thus creates a production level TLS certificate for your application using [Let's Encrypt](https://letsencrypt.org/). This happens by using the [cert-manager](https://cert-manager.io/docs/installation/kubernetes/) which is one more of the components Epinio installs with `epinio install`.
```
Comment (andreas-kupries) [inline docs/detailed-push-process.md:65]: `This happens by using the ... which` :arrow_right: `This is done by the ..., which`.
Note the `,` I added before `which`.

File: docs/detailed-push-process.md [hunk 7]
```diff
@@ -63,16 +59,13 @@ myapplication.my_epinio_system_domain.com
...
+These services resolve all subdomains of the root domain to the subdomain IP address. E.g. `1.2.3.4.omg.howdoi.website` simply resolves to `1.2.3.4`. They are useful when you don't have a real domain but you still need a wildcard domain to create subdomains to. Depending on your setup, the IP address of the cluster which Epinio discovers automatically may not be accessible by your browser and thus you may need to set the system domain when installing to use another IP. This is the case for example when you run a Kubernetes cluster with docker (e.g. [k3d](https://k3d.io/) or [kind](https://github.com/kubernetes-sigs/kind)) inside a VM (for example when using docker on Mac). Then the IP address which Epinio detects is the IP address of the docker container but that is not accessible from your host. You will need to bind the container's ports `80` and `443` to the VMs ports `80` and `443` and then use the VMs IP address instead.
```
Comment (andreas-kupries) [inline docs/detailed-push-process.md:68]: `subdomain IP address` :arrow_right: `subdomain's IP address` (added `'s` (possessive)).
`create subdomains to.` :arrow_right: `create subdomains for.`

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
A number of small nits.

File: docs/detailed-push-process.md (hunk 5 above)
Comment (jimmykarily) [reply docs/detailed-push-process.md:46]: the cert-manager is creating a certificate for the application. I didn't want to describe that here to keep the focus on how the application becomes accessible, maybe it deserves another section to explain how certs work. Maybe on another PR?

I can't think of any other resources that are created, I hope I'm not missing something.

Comment (andreas-kupries) [reply docs/detailed-push-process.md]: yes, that is ok

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**